discordbot/commands/drc.py

```diff
@@ -82,29 +96,32 @@ async def matchups(ctx: MtgContext, *, args: str) -> None:
         domain=link_domain, length=output['length'], wr=output['winrate'], url=output['compress'])
     await ctx.send(ans)
 
-def search_dreadrise(query: str) -> Tuple[int, List[str]]:
+async def search_dreadrise(query: str) -> Union[Tuple[str, None], Tuple[int, List[str]]]:
     """Returns a tuple. First member is an integer indicating how many cards match the query total,
        second member is a list of card names up to the maximum that could be fetched in a timely fashion."""
     if not query:
         return 0, []
 
     query = fetch_tools.escape(query)
     url1 = f'{domain}/cards/find?q={query}+output:resultcount'
+    count_txt = await fetch_tools.fetch_async(url1)
```

Comment:
Our "normal" usage of `fetch_tools` is to create a wrapper function in `fetcher` that uses the lower-level `fetch_tools` functions. So you'd make something like:

    def dreadrise_card_count(query: str) -> int:
        domain = configuration.get_str('dreadrise_url')
        query = fetch_tools.escape(query)
        url = f'{domain}/cards/find?q={query}+output:resultcount'
        txt = await fetch_tools.fetch_async(url)
        try:
            n = int(txt)
        except ValueError as e:
            raise FetchExecption(f'Unable to get card count with query `{query}`') from e
            
and then this whole code block would be replaced by:

    n = fetcher.dreadrise_card_count(query)
    
Notice that when you break it apart like this into well-named functions that do a single thing you then see issues like needing to have vars named `url1` and `url2` melt away.

This entire function becomes something like:

    if not query:
        return 0, [], 'no query'
    cards_fetching = DEFAULT_CARDS_SHOWN if count > MAX_CARDS_SHOWN else MAX_CARDS_SHOWN
    n = fetcher.dreadrise_card_count(query)
    cs = fetcher.dreadrise_card_search(query, True)
    if len(cs) == cards_fetching:
        return n, cs, None
    cs += fetcher.dreadrise_card_search(query, False)
        return n, cs, None

Reply:
interesting, i can do that. i don't know how async works in python, but i know dreadrise queries can take a while if they have `sf` or `url` clauses, so the `fetcher.dreadrise_card_whatever` should be async too right? otherwise the bot gets blocked

Reply:
Yeah, make them async if they are potentially long-running.

See `all_cards_async` for an example in fetcher that's already async.